File: Flyweight/PHP/PlayerFactory.php
<?php

require_once "IPlayer.php";
require_once "Terrorist.php";
require_once "CounterTerrorist.php";

class PlayerFactory {
    private array $players = [];

    public function createPlayer(string $type) {
        if (array_key_exists($type, $this->players)) {
            return $this->players[$type];
        } else {
            $player = null;

            switch ($type) {
                case 'Terrorist':
                    $player = new Terrorist;
                    $this->players[$type] = $player;
                    break;

                case 'CounterTerrorist':
                    $player = new CounterTerrorist;
                    $this->players[$type] = $player;
                    break;
            }

            return $player;
        }
    }
}

File: Flyweight/Python 3/PlayerFactory.py
from CounterTerrorist import CounterTerrorist
from Terrorist import Terrorist


class PlayerFactory:
  def __init__(self):
    self.__players = {}
  
  def createPlayer(self, type):
    if type in self.__players:
      return self.__players[type]
    else:
      player = None

      if type == "Terrorist":
        player = Terrorist()
        self.__players[type] = player
      else:
        player = CounterTerrorist()
        self.__players[type] = player
      
      return player
